job-runner/chain/chain.go

```diff
@@ -96,108 +117,167 @@ func (c *chain) NextJobs(jobId string) proto.Jobs {
 	return nextJobs
 }
 
-// JobIsReady returns whether or not a job is ready to run. A job is considered
-// ready to run if all of its previous jobs are complete. If any previous jobs
-// are not complete, the job is not ready to run.
-func (c *chain) JobIsReady(jobId string) bool {
-	c.jcMux.RLock()
-	defer c.jcMux.RUnlock()
-	for _, job := range c.previousJobs(jobId) {
-		if job.State != proto.STATE_COMPLETE {
-			return false
+// IsRunnable returns whether or not a job is runnable. A job is considered
+// runnable if it is Pending or Stopped with some retry attempts remaining,
```

Comment:
When would a Stopped job be runnable? When we make a NewChain, don't all Stopped jobs get their state changed to Pending?

Reply:
b/c of the above^ change to keep Stopped jobs Stopped instead of Pending, this can stay as-is.